jib-core/src/main/java/com/google/cloud/tools/jib/filesystem/LockFile.java

```diff
@@ -58,26 +61,31 @@ public static LockFile lock(Path lockFile) throws IOException {
     }
 
     Files.createDirectories(lockFile.getParent());
-    FileLock fileLock = new FileOutputStream(lockFile.toFile()).getChannel().lock();
-    return new LockFile(lockFile, fileLock);
+    FileOutputStream outputStream = new FileOutputStream(lockFile.toFile());
+    FileLock fileLock = null;
+    try {
+      fileLock = outputStream.getChannel().lock();
+      return new LockFile(lockFile, fileLock, outputStream);
+
+    } finally {
+      if (fileLock == null) {
+        outputStream.close();
+      }
+    }
   }
 
   /** Releases the lock file. */
   @Override
   public void close() {
     try {
       fileLock.release();
+      outputStream.close();
```

Comment:
I missed this fun somehow yesterday, but closing the stream closes the channel, and closing the channel releases the lock.

Reply:
Yeah, we figured that out, but we decided to release it explicitly as we hold both references anyways and to eliminate a potential doubt for future readers.